JavaScript/asynchronousQueue.js

```diff
@@ -0,0 +1,210 @@
+      console.log(color.green + 'Factors of this Queue: ' +
+        this.factorsOfQueue.join(', ') + color.white + '\n' + SEPARATOR) :
+      console.log(color.green + 'This Queue does not have factors' +
+        color.white + '\n' + SEPARATOR);
+    return this;
+  }
+
+  _applyFactors(parameter, iterable, array) {
+    iterable.forEach((value, index) => {
+      if (this._canBePushed(parameter, value)) array.push(this.users[index]);
+    });
+    return array.length;
+  }
+
+  _canBePushed(parameter, value) {
+    return typeof parameter === 'string' && parameter.includes(value);
+  }
+
+  _useFactors() {
+    for (const value of this.factorsOfQueue) {
+      const index = this.users.indexOf(value);
+      const elements = this.arraysOfTasks[index];
+      const length = elements.length;
+      this.queue.push(...elements);
+      this.timeouts.push(...Array(length).fill(0.5));
+      this.usersByTasks.push(...Array(length).fill(this.users[index]));
+      this.priorities[index] = 0;
+    }
+  }
+
+  _usePriorities() {
+    let timeout = 0;
+    for (let i = 0; i < this.priorities.length; i++) {
+      const max = this._maxElement(this.priorities);
+      if (max) {
+        timeout++;
+        this._findPriority(timeout, max);
+      }
+    }
+  }
+
+  _maxElement(array) {
+    return Math.max(...array);
+  }
+
+  _findPriority(timeout, max) {
+    const indexOfMax = this.priorities.indexOf(max);
+    const user = this.users[indexOfMax];
+    const userData = Array.from(this.data.keys());
+    userData.filter(object => object.user === user)
+      .map(object => this._addElementsToCollections(this.data.get(object), user, timeout));
+  }
+
+  _noPriorities() {
+    let timeout = 0;
+    for (const tasks of this.arraysOfTasks) {
+      const indexOfValue = this.arraysOfTasks.indexOf(tasks);
+      if (this.priorities[indexOfValue]) {
+        timeout++;
+        const user = this.users[indexOfValue];
+        this._addElementsToCollections(tasks, user, timeout);
+      }
+    }
+  }
+
+  _addElementsToCollections(tasks, user, timeout) {
+    const index = this.arraysOfTasks.indexOf(tasks);
+    this.queue.push(...tasks);
+    const length = tasks.length;
+    this.usersByTasks.push(...Array(length).fill(user));
+    const intervals = Array(length).fill(timeout);
+    this.priorities[index] = 0;
+    if (length > 1) this._setIntervalsBetweenTasks(intervals, length);
+    this.timeouts.push(...intervals);
+  }
+
+  _setIntervalsBetweenTasks(intervals, length) {
+    const difference = 1 / length;
+    for (let i = 0; i < length; i++) {
+      intervals[i] += i * difference;
+    }
+  }
+
+  init() {
+    this.formed = true;
+    if (this.factorsOfQueue.length) this._useFactors();
+    if (this.prioritised) this._usePriorities();
+    if (!this.prioritised) this._noPriorities();
+    return this;
+  }
+
+  async start() {
+    if (!this.pushed) throw new Error(color.red +
+      'Queue is empty. You must add the elements first!' + color.white);
```

Comment:
```suggestion
    if (!this.pushed) {
      throw new Error(
        color.red + 'Queue is empty. You must add the elements first!' + color.white
      );
    }
```
It's not a good idea to put colors into error message